### PR TITLE
API modal

### PR DESCRIPTION
**Pull request by Jin1L**
Merged all the API setting UI components into one.
Updated setter and getter API functions to create and use the `credential.yaml` file.
Updated the Visual Tested according to the changes that I made in UI components. Deleted `S3-key-test`, testing everything in `set-api-test`.

image

### CODE REVIEW

**Comment by yassine-moumen** [comment]
## To access Flojoy-studio build from this pull request, you can use the following URLs: 

Windows installation: http://660_windows.flojoy-test.com:3000 
Linux installation: http://660_linux.flojoy-test.com:3000 

Please note that it may take a few minutes for the instance to start up, so please be patient. 

It is important to know that the instance will automatically shut down after 4 hours to conserve resources. If you wish to start the instance again after it shuts down (or restart it to pickup new changes), you can use this API : 

Windows: https://kgpsoscpc4q52d4mtwuu7jhvie0qdiso.lambda-url.us-east-1.on.aws/start_instance?instance_id=i-0d7256f30cf567f35&domain=660_windows 

Linux: https://kgpsoscpc4q52d4mtwuu7jhvie0qdiso.lambda-url.us-east-1.on.aws/start_instance?instance_id=i-0bcf8fdeded1d4f87&domain=660_linux

File: src/feature/flow_chart_panel/views/APIKeyModal.tsx [hunk 1]
```diff
@@ -75,20 +83,89 @@ const useStyles = createStyles((theme) => ({
...
+    cloudApiKey,
+    setCloudApiKey,
+    openAIApiKey,
+    setOpenAIApiKey,
+    s3Container,
+    setS3Container,
+  } = useFlowChartState();
+  const { classes } = useStyles(s3Container);
+  const [s3Name, setS3Name] = useState<string>("");
+  const { s3AccessKey, setS3AccessKey, s3SecretKey, setS3SecretKey } =
```
Comment (JeffDotPng) [inline src/feature/flow_chart_panel/views/APIKeyModal.tsx:126]: duplicate hook call?

File: src/services/FlowChartServices.ts [hunk 2]
```diff
@@ -27,9 +27,44 @@ export function saveFlowChartToLocalStorage(rfInstance?: ReactFlowJsonObject) {
...
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ key: apiKey }),
+    });
+
+    if (response.ok) {
+      const responseData = await response.json();
+      notifications.update({
+        id: "set-api-key",
+        title: "Successful!",
```
Comment (JeffDotPng) [inline src/services/FlowChartServices.ts:44]: The 3 functions to send the post request are almost exactly the same except for the endpoint they query, please refactor

**Review by JeffDotPng — CHANGES_REQUESTED** [review]

File: src/feature/flow_chart_panel/views/APIKeyModal.tsx [hunk 3]
```diff
@@ -75,20 +82,94 @@ const useStyles = createStyles((theme) => ({
...
+  } = useFlowChartState();
+  const { classes } = useStyles(s3Container);
+  const {
+    s3Name,
+    setS3Name,
+    s3AccessKey,
+    setS3AccessKey,
+    s3SecretKey,
+    setS3SecretKey,
+  } = useFlowChartState();
```
Comment (JeffDotPng) [inline src/feature/flow_chart_panel/views/APIKeyModal.tsx:131]: still duplicate hook call

File: src/services/FlowChartServices.ts [hunk 4]
```diff
@@ -27,9 +27,9 @@ export function saveFlowChartToLocalStorage(rfInstance?: ReactFlowJsonObject) {
   }
 }
 
-export const sendApiKeyToDjango = async (apiKey: string) => {
+export const sendApiKeyToDjango = async (apiKey: string, apiFor: string) => {
```
Comment (JeffDotPng) [inline src/services/FlowChartServices.ts:30]:
```typescript
export const postApiKey = async (endpoint: string, body: object) => {
  try {
    const response = await fetch(`${API_URI}/api/${endpoint}`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(body),
    });

    if (response.ok) {
      const responseData = await response.json();
      notifications.update({
        id: "set-api-key",
        title: "Successful!",
        message: "Successfully set key",
        autoClose: 5000,
      });
    } else {
      notifications.update({
        id: "set-api-key",
        title: "Failed!",
        message: "Failed to set key",
        autoClose: 5000,
      });
    }
  } catch (error) {
    notifications.update({
      id: "set-api-key",
      title: "Failed!",
      message: "Failed to set key",
      autoClose: 5000,
    });
  }
};

export const postCloudApiKey = async (key: string) => {
  await postApiKey("set-cloud-key", { key });
};

export const postS3Key = async (
  s3Name: string,
  s3AccessKey: string,
  s3SecretKey: string
) => {
  await postApiKey("set-s3-key", {
    name: s3Name,
    accessKey: s3AccessKey,
    secretKey: s3SecretKey,
  });
};

export const postOpenAIKey = async (key: string) => {
  await postApiKey("set-openai-key", { key });
};
```

**Review by JeffDotPng — CHANGES_REQUESTED** [review]

File: src/services/FlowChartServices.ts [hunk 5]
```diff
@@ -27,14 +27,14 @@ export function saveFlowChartToLocalStorage(rfInstance?: ReactFlowJsonObject) {
   }
 }
 
-export const sendApiKeyToDjango = async (apiKey: string) => {
+export const sendApiKeyToDjango = async (body: object, endpoint: string) => {
```
Comment (itsjoeoui) [inline src/services/FlowChartServices.ts:30]: type this object

**Review by JeffDotPng — APPROVED** [review]

**Pull request merged**